File: pythonPractice/Arrays/MergedSortedArrays.py
def merge_sorted_arrays(arr1,arr2):
		i, j = 0 , 0
		merged = []
		while i < len(arr1) and j < len(arr2):
			if(arr1[i] < arr2[j]):
				merged.append(arr1[i]) 
				i += 1
			else:
				merged.append(arr2[j])
				j += 1
		
		merged.extend(arr1[i:])
		merged.extend(arr2[j:])

		return merged
	


arr1 = [1,3,4,5,6]
arr2 = [8,9,16,20]
print(merge_sorted_arrays(arr1,arr2))

File: javaPractice/ArrayKa/MergedSortedArrays.java
import java.util.Arrays;

public class MergedSortedArrays{

		public static int [] mergeSortedArrays(int[] arr1,int [] arr2){
			int [] result = new int[arr1.length + arr2.length];


			int i = 0;
			int j = 0;
			int k = 0;


			while (i < arr1.length && j < arr2.length){
				if(arr1[i] < arr2[j]) {
					result[k++] = arr1[i++];
				}else{
					result[k++] = arr2[j++];
				}
			}
				while (i < arr1.length){
					result[k++] = arr1[i++];
				}

				while (j < arr2.length){
					result[k++] = arr2[j++];
				}

				return result;
				}


public static void main(String[] args){

		int[] arr1 = {1,3,5,7};
		int[] arr2 = {2,4,6,8};

		int [] merged = mergeSortedArrays(arr1,arr2);

	System.out.println("The Merged Array is : " + Arrays.toString(merged));

		}
	}